modules/tensor_mechanics/include/userobjects/TensorMechanicsPlasticIsotropicSD.h

```diff
@@ -0,0 +1,81 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef TENSORMECHANICSPLASTICISOTROPICSD_H
+#define TENSORMECHANICSPLASTICISOTROPICSD_H
+
+#include "TensorMechanicsPlasticJ2.h"
+
+class TensorMechanicsPlasticIsotropicSD;
+
+template<>
+InputParameters validParams<TensorMechanicsPlasticIsotropicSD>();
+
+/**
+ * IsotropicSD plasticity model from Yoon (2013)
+ * the name of the paper is "Asymmetric yield function based on the
+ * stress invariants for pressure sensitive metals" published
+ * 4th December 2013.
+ * This model accounts for sensitivity in pressure and for the
+ * strength differential effect
+ * Yield_function = a[b*I1 + (J2^{3/2} - c*J3)^{1/3}] - yield_strength
+ * The last three functions are the main functions that call all other
+ * functions in this module for the Newton-Raphson method.
+ */
+class TensorMechanicsPlasticIsotropicSD : public TensorMechanicsPlasticJ2
+{
+public:
+  TensorMechanicsPlasticIsotropicSD(const InputParameters & parameters);
+protected:
+
+  const Real _b;
+
+  Real _c;
+
+  const bool _associative;
+
+  RankFourTensor _h;
+
+  Real _a;
+
+  //derivative of phi with respect to J2, phi is b*I1 + (J2^{3/2} - c*J3)^{1/3}
+  Real  dphi_dj2(const Real j2, const Real j3)const;
+
+  // derivative of phi with respect to J3
+  Real  dphi_dj3(const Real j2, const Real j3)const;
+
+  // derivative of dphi_dJ2 with respect to J2
+  Real  dfj2_dj2(const Real j2, const Real j3)const;
+
+  // derivative of dphi_dJ2 with respect to J3
+  Real  dfj2_dj3(const Real j2, const Real j3)const;
+
+  // derivative of dphi_dJ3 with respect to J2
+  Real  dfj3_dj2(const Real j2, const Real j3)const;
+
+  // derivative of dphi_dJ3 with respect to J3
+  Real  dfj3_dj3(const Real j2, const Real j3)const;
+
+  // derivative of the trace with respect to sigma rank two tensor
+  RankTwoTensor dI_sigma()const;
+
+  // derivative of the second invariant with respect to the stress deviatoric tensor
+  RankTwoTensor dj2_dSkl(const RankTwoTensor & stress)const;
+
+  //Yield_function = a[b*I1 + (J2^{3/2} - c*J3)^{1/3}] - yield_strength
+  Real yieldFunction(const RankTwoTensor & stress, Real intnl) const override;
+
+  //Tensor derivative of the yield_function with respect to the stress tensor
+  RankTwoTensor dyieldFunction_dstress(const RankTwoTensor & stress, Real intnl) const override;
+
+  //Tensor derivative of the tensor derivative of the yield_function with respect to the stress tensor
+  RankFourTensor dflowPotential_dstress(const RankTwoTensor & stress, Real intnl) const override;
+
+  RankTwoTensor flowPotential(const RankTwoTensor & stress, Real intnl) const override;
+
```

Comment:
empty line.

I suggest you go over the PR yourself again and look for those issues.